### PR TITLE
Invoice generation logic

### PR DESCRIPTION
**Pull request by GauravGusain98**
### changes
- Added billing level option while generating the invoice
- Create migration to make project id nullable in invoice
- Update the code everywhere to handle the null project id in invoice
- Updated the name of project  with null project id to client name + "Projects"

### CODE REVIEW

**Review by P4NK4J — APPROVED** [review]

**Review by kuldeep3 — APPROVED** [review]
Few suggestions. Please make changes based on our offline conversation 🔧

File: Modules/Invoice/Http/Controllers/InvoiceController.php [hunk 1]
```diff
@@ -180,4 +203,12 @@ public function sendEmail(Request $request, Invoice $invoice, Client $client)
 
         return redirect(route('invoice.index'));
     }
+
+    public function sendInvoice(Request $request)
+    {
+        $client = Client::find($request->client_id);
+        $this->service->sendInvoice($client, $request->term, $request->all());
+
+        return redirect()->back()->with('status', 'Invoice sent successfully.');
```
[the remark this answers is not in the document]
Comment (GauravGusain98) [reply Modules/Invoice/Http/Controllers/InvoiceController.php:212]: Update the message.

File: Modules/Invoice/Resources/views/index.blade.php [hunk 2]
```diff
@@ -79,7 +86,7 @@
                     </tr>
                 </thead>
                 <tbody>
-                    @if((request()->invoice_status == 'sent' || $invoiceStatus == 'sent'))
+                    @if((request()->invoice_status == 'sent' || $invoiceStatus == 'sent') && !$invoices->isEmpty())
```
[the remark this answers is not in the document]
Comment (GauravGusain98) [reply Modules/Invoice/Resources/views/index.blade.php:89]: done

File: Modules/Invoice/Resources/views/index.blade.php [hunk 3]
```diff
@@ -135,11 +142,114 @@
                                 @endif
                             </tr>
                         @endforeach
+                    @elseif((request()->invoice_status == 'ready'  || $invoiceStatus == 'ready') && !$readyToSendInvoicesData->isEmpty())
```
[the remark this answers is not in the document]
Comment (GauravGusain98) [reply Modules/Invoice/Resources/views/index.blade.php:145]: done

File: Modules/Invoice/Resources/views/index.blade.php [hunk 4]
```diff
@@ -135,11 +142,114 @@
...
+                        @foreach ($readyToSendInvoicesData as $client)
```
[the remark this answers is not in the document]
Comment (GauravGusain98) [reply Modules/Invoice/Resources/views/index.blade.php:146]: done

File: Modules/Invoice/Resources/views/index.blade.php [hunk 5]
```diff
@@ -135,11 +142,114 @@
...
+                                    <label class="leading-none" for="sendToName">Receiver Name</label>
+                                    <input type="text" name="to_name" id="sendToName"
+                                        class="form-control">
+                                </div>
+                                <div class="form-group col-md-12">
+                                    <label class="leading-none" for="cc">
+                                        CC 
+                                        <span data-toggle="tooltip" data-placement="right" title="Comma separated emails">
+                                            <i class="fa fa-question-circle"></i>
+                                        </span></label>
```
[the remark this answers is not in the document]
Comment (GauravGusain98) [reply Modules/Invoice/Resources/views/index.blade.php:223]: done

**Pull request merged**